### PR TITLE
Wimpie/employee address service refactor

### PR DESCRIPTION
**Pull request by Wimmas**
image

### CODE REVIEW

**Review by Juanro-Minne — APPROVED** [review]

File: HRIS.Services/Interfaces/IEmployeeAddressService.cs [hunk 1]
```diff
@@ -9,21 +9,21 @@ public interface IEmployeeAddressService
     /// </summary>
     /// <param name="employeeAddressDto"></param>
```
Comment (EduanO) [inline HRIS.Services/Interfaces/IEmployeeAddressService.cs]:
```suggestion
    /// <param name="id"></param>
```

File: HRIS.Services/Interfaces/IEmployeeAddressService.cs [hunk 2]
```diff
@@ -9,21 +9,21 @@ public interface IEmployeeAddressService
...
     /// <returns>Employee Address</returns>
-    Task<EmployeeAddressDto> Save(EmployeeAddressDto employeeAddressDto);
+    Task<EmployeeAddressDto> Create(EmployeeAddressDto employeeAddressDto);
 
     /// <summary>
     ///     Get Employee Address
     /// </summary>
     /// <param name="employeeId"></param>
```
Comment (EduanO) [inline HRIS.Services/Interfaces/IEmployeeAddressService.cs]:
```suggestion
    /// <param name="id"></param>
```

File: HRIS.Services/Interfaces/IEmployeeAddressService.cs [hunk 3]
```diff
@@ -9,21 +9,21 @@ public interface IEmployeeAddressService
...
     /// <returns>Employee Address</returns>
-    Task<EmployeeAddressDto> Get(EmployeeAddressDto employeeAddressDto);
+    Task<EmployeeAddressDto> Get(int id);
```
Comment (EduanO) [inline HRIS.Services/Interfaces/IEmployeeAddressService.cs]:
```suggestion
    Task<EmployeeAddressDto> GetById(int id);
```

**Review by EduanO — CHANGES_REQUESTED** [review]

**Review by madwizzard — APPROVED** [review]

File: HRIS.Services.Tests/Services/EmployeeAddressServiceUnitTest.cs [hunk 4]
```diff
@@ -4,6 +4,7 @@
 using Moq;
 using RR.Tests.Data;
 using RR.UnitOfWork;
+using RR.UnitOfWork.Entities.ATS;
```
Comment (bseptember) [inline HRIS.Services.Tests/Services/EmployeeAddressServiceUnitTest.cs]: Refactor: we don't need these lines:
using HRIS.Services.Interfaces;
using RR.UnitOfWork.Entities.ATS;

**Review by bseptember — CHANGES_REQUESTED** [review]
@Wimmas there is no validation added on any of these? existence must be checked, then is the authority correct to perform the service method.
eg.
if (_identity.IsSupport == false)
            throw new CustomException("Unauthorized Access.");

**Review by EduanO — APPROVED** [review]

File: HRIS.Services.Tests/Services/EmployeeAddressServiceUnitTest.cs (hunk 4 above)
Comment (Wimmas) [reply HRIS.Services.Tests/Services/EmployeeAddressServiceUnitTest.cs]: Okay cool
Will add the auth stuff

**Comment by bseptember** [comment]
> @Wimmas there is no validation added on any of these? existence must be checked, then is the authority correct to perform the service method. eg. if (_identity.IsSupport == false) throw new CustomException("Unauthorized Access.");

@Wimmas @EduanO The logic is not correct in your recent [commit ](https://github.com/RetroRabbit/RGO-Server/pull/218/commits/b9ff4e4388d8471b45b6b339af3af30eb58f9ce6)

there are some methods that need the _identity.id == parameter_id compared and the identity support check.

File: HRIS.Services/Services/EmployeeAddressService.cs [hunk 5]
```diff
@@ -34,6 +37,9 @@ public async Task<EmployeeAddressDto> Delete(int addressId)
             throw new CustomException("Employee Address Does Not Exist");
         }
 
+        if (_identity.IsSupport == false)
```
Comment (bseptember) [inline HRIS.Services/Services/EmployeeAddressService.cs:40]: Fix: this here is not necessary

Comment (bseptember) [inline HRIS.Services/Services/EmployeeAddressService.cs]: Logic: missing condition here

File: HRIS.Services/Services/EmployeeAddressService.cs [hunk 6]
```diff
@@ -47,12 +53,18 @@ public async Task<EmployeeAddressDto> GetById(int id)
...
```
Comment (bseptember) [inline HRIS.Services/Services/EmployeeAddressService.cs]: logic: missing cond.

File: HRIS.Services/Services/EmployeeAddressService.cs [hunk 7]
```diff
@@ -65,6 +77,9 @@ public async Task<EmployeeAddressDto> Create(EmployeeAddressDto employeeAddressD
             throw new CustomException("Employee Address Already Exists");
         }
 
+        if (_identity.IsSupport == false)
```
Comment (bseptember) [inline HRIS.Services/Services/EmployeeAddressService.cs]: logic: missing cond.

File: HRIS.Services/Services/EmployeeAddressService.cs [hunk 8]
```diff
@@ -79,6 +94,9 @@ public async Task<EmployeeAddressDto> Update(EmployeeAddressDto employeeAddressD
             throw new CustomException("Employee Address Does Not Exist");
         }
 
+        if (_identity.IsSupport == false)
```
Comment (bseptember) [inline HRIS.Services/Services/EmployeeAddressService.cs]: logic: missing cond.

**Review by bseptember — CHANGES_REQUESTED** [review]

**Review by bseptember — APPROVED** [review]

**Pull request merged**